syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.snippet_sample";
option java_outer_classname = "snippet_sample_Proto";
option objc_class_prefix = "SSP";

package StorageEngineInstance;

service LBA2PBAManager {
  rpc RequestPBA (LBARequest) returns (PBAResponse) {}
}

message LBARequest{
  message LBA {
    string sst_name = 1;
    string csd_id = 2;
    repeated Chunk chunks = 3;
  }
  repeated LBA lba_chunks = 1;
}

message Chunk {
  int64 offset = 1;
  int32 length = 2;
}

message PBAResponse{
  message PBA{
    repeated Chunk chunks = 1;
  }
  map<string,PBA> pba_chunks = 1;
}